homeassistant/components/matter/light.py

```diff
@@ -39,79 +75,305 @@ class MatterLight(MatterEntity, LightEntity):
 
     entity_description: MatterLightEntityDescription
 
+    def _supports_feature(
+        self, feature_map: int, feature: MatterColorControlFeatures
+    ) -> bool:
+        """Return if device supports given feature."""
+
+        return (feature_map & (1 << feature.value)) != 0
+
+    def _supports_color_mode(self, color_feature: MatterColorControlFeatures) -> bool:
+        """Return if device supports given color mode."""
+
+        feature_map = self._device_type_instance.node.get_attribute(
+            self._device_type_instance.endpoint,
+            clusters.ColorControl,
+            clusters.ColorControl.Attributes.FeatureMap,
+        )
+
+        assert isinstance(feature_map.value, int)
```

Comment:
Why do we assert here? Is it to satisfy mypy?

Reply:
That is indeed to satisfy mypy